support for setting/sending multiple cookies

StevenVanAcker opened the pull request:
This patch adds support for using multiple cookies set by the same server, when using stickycookies.
The original code only remembered the last cookie set.

Testcase through: mitmdump -t "."
visit https://httpbin.org/cookies/set?helloWWW=worldZZZ&xxx=yyy
then flush cookies from your browser
and finally visit https://httpbin.org/cookies

mhils commented on mitmproxy/flow.py:
```diff
@@ -337,7 +337,9 @@ def handle_response(self, f):
                 if self.domain_match(f.request.host, k[0]):
-                    self.jar[k] = m
+                    if not k in self.jar:
+                        self.jar[k] = {}
```
Could we use `collections.defaultdict` instead? :smiley:

mhils commented on mitmproxy/flow.py:
```diff
@@ -349,10 +351,10 @@ def handle_request(self, f):
                 ]
                 if all(match):
-                    l.append(self.jar[i].output(header="").strip())
+                    l.extend([m.output(header="").strip() for (_, m) in self.jar[i].items()])
```
Very minor nitpick: Why not just `.values()`?

mhils commented:
Thanks for the excellent PR! :smiley:
It'd be great if you could add a small test at [test/mitmproxy/test_flow.py](https://github.com/mitmproxy/mitmproxy/blob/master/test/mitmproxy/test_flow.py#L45), but we can take that over if you are busy with other things.

StevenVanAcker commented:
I've switched to defaultdict, simplified the list comprehension with .values() and squashed the new commits into one. I'll have a look at the test as well later today

mhils commented:
Awesome. Thanks so much for the great PR! :smiley: :cake: :tada:
